scoped/scoped.go

```
@@ -0,0 +1,42 @@
+package scoped
+
+import (
+	ds "github.com/ipfs/go-datastore"
+)
+
+func getFeatureSet(dstore ds.Datastore) map[interface{}]bool {
+	features := ds.FeaturesForDatastore(dstore)
+	featureSet := map[interface{}]bool{}
+	for _, f := range features {
+		featureSet[f.Interface] = true
+	}
+	return featureSet
+}
+
+// Wrap returns a datastore based on the source, whose concrete type is scoped down to only the features supported by the target.
```

This sentence is wrong, as far as I can tell, this actually return the intersection of features.

I think the intersection is less useful, and I think it should error instead of trying to silently continue.

---

The elements of set A that are also in set B is the definition of set intersection, the two statements are equivalent.

The reason this doesn't return an error if B is not a subset of A is to allow for both datastores to evolve independently, without breaking consumers of this.

---

For me this sentence assume that set A always is >= than set B. It's not really important.